state-chain/pallets/cf-swapping/src/lib.rs

```diff
@@ -145,19 +147,22 @@ impl SwapState {
 }
 
 #[repr(u8)]
-#[derive(Clone, PartialOrd, Ord, Debug, PartialEq, Eq, Encode, Decode, TypeInfo)]
-enum FeeType {
+#[derive(Clone, DebugNoBound, PartialEq, Eq, Encode, Decode, TypeInfo)]
+#[scale_info(skip_type_params(T))]
+enum FeeType<T: Config> {
 	NetworkFee = 0,
+	BrokerFee(Beneficiaries<T::AccountId>),
```

Comment:
I don't see how changing the structure changes anything tbh. We could already apply this in parallel with the existing structure. In fact this was my frist implementation, but then I decided against it since it could lead to some problematic scenarios and error prone code. For example, we need to ensure that broker fees + network fee (plus whatever other fee we might add in the future) do not exceed 100% since otherwise we would be creating money out of thin air. Lots of opportunity for error for very little gain. I also went with the sequential approach since that's the existing behaviour, and we even agreed on this on discord.